server/src/services/address.service.ts

```diff
@@ -0,0 +1,98 @@
+import User from '../models/user.model';
+import UserAddress from '../models/userAddress.model';
+
+import { ICreateAddress, IUpdateAddress } from '../interfaces/addressService.interface';
+
+class AddressService {
+  private async _updateUserDefaultAddressByUserId(
+    transaction: Transaction,
+    user_id: string,
+    address_id: number | null
+  ) {
+    const [userUpdateSuccess] = await User.update(
+      { default_address: address_id },
+      {
+        where: {
+          id: user_id,
+        },
+        transaction,
+      }
+    );
+    if (!userUpdateSuccess) {
+      throw Error(); // unexpected error
+    }
+  }
+
+  public async createAddress(transaction: Transaction, user_id: string, address: ICreateAddress): Promise<UserAddress> {
+    const isDefaultAddress: boolean | undefined = address.default_address;
+    const newUserAddress: UserAddress = (await UserAddress.create({ ...address, user_id }, { transaction })).get();
+
+    if (isDefaultAddress) {
+      await this._updateUserDefaultAddressByUserId(transaction, user_id, newUserAddress.id);
+    }
+    return newUserAddress;
+  }
+
+  public async getAddressesByUserId(user_id: string): Promise<UserAddress[]> {
+    const results: UserAddress[] = await UserAddress.findAll({
+      where: {
+        user_id,
+      },
+      attributes: ['id', 'name', 'address', 'phone', 'zip_code'],
+    });
+
+    return results;
+  }
+
+  public async updateAddressById(transaction: Transaction, user_id: string, address: IUpdateAddress): Promise<void> {
+    // check this address_id is owner's default_address
+    const isDefaultAddress =
+      (await User.findOne({ where: { id: user_id, default_address: address.id } })) !== null ? true : false;
+
+    const default_address: boolean | undefined = address.default_address;
+    const [updateSuccess] = await UserAddress.update(
+      { ...address },
+      {
+        where: {
+          id: address.id,
+          user_id,
+        },
+        transaction,
+      }
+    );
+    if (!updateSuccess) {
+      // address id not match
+      throw new Error(); // unexpected Error
+    }
+
+    if (default_address) {
+      await this._updateUserDefaultAddressByUserId(transaction, user_id, address.id);
+    } else if (isDefaultAddress) {
+      await this._updateUserDefaultAddressByUserId(transaction, user_id, null);
+    }
+  }
+
+  public async deleteAddressById(transaction: Transaction, user_id: string, id: number): Promise<void> {
+    const isDefaultAddress =
+      (await User.findOne({ where: { id: user_id, default_address: id } })) !== null ? true : false;
+
+    const deleteCount = await UserAddress.destroy({
+      where: {
+        id,
+        user_id,
+      },
+      transaction,
+    });
+
+    if (deleteCount >= 1) {
+      if (isDefaultAddress) {
+        await this._updateUserDefaultAddressByUserId(transaction, user_id, null);
+      }
+      return;
+    }
+    throw new Error(); // unexpected Error
```

Comment:
```suggestion
    if (deleteCount !== 1) {
      throw new Error(); // unexpected Error
    }
    const isDefaultAddress =
      (await User.findOne({ where: { id: user_id, default_address: id } })) !== null ? true : false;
    if (isDefaultAddress) {
      await this._updateUserDefaultAddressByUserId(transaction, user_id, null);
    }
```

if문의 depth를 깊게 하는 것보단 최대한 하나의 depth만 갖도록 분리해보는 건 어떨까요?
또한 변수는 최대한 사용처에 가까우면 더 좋을 것 같아요 ☺️